src/lib.rs

```diff
@@ -336,77 +340,39 @@ impl Filter {
     }
 
     /// Create a new instance from a given set of filter creteria.
-    pub fn new(matches: &getopts::Matches) -> Self {
+    pub fn new(args: Args) -> Self {
```

Comment:
The disadvantage of borrowing is, that all the vectors, and the strings stored in those vectors need to be cloned, even though, that is actually not necessary. But I try to think of a better solution.

Reply:
How about explicit lifetime annotation and references to the fields in `Args`?  The `Args` instance definitely outlives the creation of the `Filter` when applying the suggestion above.

Reply:
Hmm, maybe, but this would definitely be a big overhaul then.. I think I have a easier solution, that would be similar to `Args::take_output`, that should be good enough for the time being.

Reply:
Okay, this was actually a lot easier, since only the vectors needed to be references and not the strings as well. So this has been done with references, also because the other way did not work for some reason.. 😅